datamaker/core/storage_providers/amazon_s3.py

```diff
@@ -0,0 +1,28 @@
+import boto3
+import s3path
+
+from datamaker.core.storage_providers import BaseProvider
+
+
+class AmazonS3StorageProvider(BaseProvider):
+    label = 'Amazon S3'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.s3_init = False
+
+    def _get_s3_path(self):
+        if not self.s3_init:
+            boto3_configs = {
+                'aws_access_key_id': self.configuration['access_key'],
+                'aws_secret_access_key': self.configuration['secret_key'],
+                'region_name': self.configuration['region_name'],
+            }
+            boto3.setup_default_session(**boto3_configs)
+            self.s3_init = True
+        return s3path.S3Path('/' + self.configuration['bucket_name'])
+
+    def get_pathlib(self):
+        pathlib_object = self._get_s3_path()
+        return pathlib_object
```

Comment:
[P4]
은정님 단골 스타일이긴 한데 불필요한 변수는 필요 없을 것 같습니다.

그리고 _get_s3_path를 별도로 구현할 필요 없이 그냥 get_pathlib에 구현해도 무방할 듯 합니다.

```suggestion
        return self._get_s3_path()
```